tools/optimizer/optimizer.cpp

```diff
@@ -716,7 +716,7 @@ Ref simplifyCondition(Ref node) {
 
 StringSet ELIMINATION_SAFE_NODES("assign call if toplevel do return label switch binary unary-prefix"); // do is checked carefully, however
 StringSet IGNORABLE_ELIMINATOR_SCAN_NODES("num toplevel string break continue dot"); // dot can only be STRING_TABLE.*
-StringSet ABORTING_ELIMINATOR_SCAN_NODES("new object function defun for while array throw"); // we could handle some of these, TODO, but nontrivial (e.g. for while, the condition is hit multiple times after the body)
+StringSet ABORTING_ELIMINATOR_SCAN_NODES("new object defun while array"); // we could handle some of these, TODO, but nontrivial (e.g. for while, the condition is hit multiple times after the body)
```

Comment:
Hmm - that check should indeed not be needed. The heap replacer method for memory growth should not be seen by the asm.js optimizer. But we should verify that. If you don't see an error removing that line (and the `ABORTING_` stuff) then it's fine.